File: B62722/Laboratorios/LaboratorioLibreria/Orbis/src/main/java/com/orbis/b62722/domain/Libro.java
package com.orbis.b62722.domain;

import java.util.LinkedList;
import java.util.List;

public class Libro {
	
	private int id_libro;
	private String titulo;
	private int ano;
	private float precio;
	private Editorial editorial;
	private List<Autor> autores;
	
	public Libro() {
		autores = new LinkedList<Autor>();
		editorial = new Editorial();
	}

	public int getId_libro() {
		return id_libro;
	}

	public void setId_libro(int id_libro) {
		if(id_libro <= 0) throw new RuntimeException("El id debe ser mayor a 0.");
		this.id_libro = id_libro;
	}

	public String getTitulo() {
		return titulo;
	}

	public void setTitulo(String titulo) {
		if(titulo == null || titulo.trim().equals("")) throw new RuntimeException("El título es requerido");
		this.titulo = titulo;
	}

	public int getAno() {
		return ano;
	}

	public void setAno(int ano) {
		this.ano = ano;
	}

	public float getPrecio() {
		return precio;
	}

	public void setPrecio(float precio) {
		if(precio <= 0) throw new RuntimeException("El precio debe ser mayor a 0.");
		this.precio = precio;
	}

	public Editorial getEditorial() {
		return editorial;
	}

	public void setEditorial(Editorial editorial) {
		this.editorial = editorial;
	}

	public List<Autor> getAutores() {
		return autores;
	}

	public void setAutores(List<Autor> autores) {
		this.autores = autores;
	}
}

File: B67156/Lab´s/Laboratorio 4.1/OrbisB67156.Web/Orbis[B67156].Model/Domain/Libro.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Orbis_B67156_.Model.Domain
{
    public class Libro
    {

        private int id_libro;
        public int Id_libro
        {
            get
            {
                return id_libro;
            }
            set
            {
                if (value <= 0) throw new Exception("Id libro es requerido");
                id_libro = value;
            }
        }
        private string titulo;
        public string Titulo
        {
            get
            {
                return titulo;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Titulo es requerido");
                titulo = value;
            }
        }
        private int año;
        public int Año
        {
            get
            {
                return año;
            }
            set
            {
                if (value <= 0) throw new Exception("Año es requerido");
                año = value;
            }
        }
        private float precio;
        public float Precio
        {
            get
            {
                return precio;
            }
            set
            {
                if (value <= 0) throw new Exception("Precio es requerido");
                precio = value;
            }
        }
        private int id_editorial;
        public int Id_editorial
        {
            get
            {
                return id_editorial;
            }
            set
            {
                if (value <= 0) throw new Exception("Id editorial es requerido");
                id_editorial = value;
            }
        }


        public Libro(int id_libro, string titulo, int año, float precio, int id_editorial)
        {
            Id_libro = id_libro;
            Titulo = titulo;
            Año = año;
            Precio = precio;
            Id_editorial = id_editorial;
        }
    }
}
